nixos/server/default.nix

```diff
@@ -1,89 +1,95 @@
 # A default configuration that applies to all servers.
 # Common configuration accross *all* the machines
-{ pkgs, lib, ... }:
+{ options, pkgs, lib, ... }:
 {
 
   imports = [
     ../common
   ];
 
-  # List packages installed in system profile.
-  environment.systemPackages = map lib.lowPrio [
-    pkgs.curl
-    pkgs.dnsutils
-    pkgs.gitMinimal
-    pkgs.htop
-    pkgs.jq
-    pkgs.tmux
-  ];
-
-  # Notice this also disables --help for some commands such es nixos-rebuild
-  documentation.enable = lib.mkDefault false;
-  documentation.info.enable = lib.mkDefault false;
-  documentation.man.enable = lib.mkDefault false;
-  documentation.nixos.enable = lib.mkDefault false;
-
-  # No need for fonts on a server
-  fonts.fontconfig.enable = lib.mkDefault false;
-
-  programs.vim.defaultEditor = lib.mkDefault true;
-
-  # Print the URL instead on servers
-  environment.variables.BROWSER = "echo";
-
-  # Make sure firewall is enabled
-  networking.firewall.enable = true;
-
-  # Delegate the hostname setting to dhcp/cloud-init by default
-  networking.hostName = lib.mkDefault "";
-
-  # If the user is in @wheel they are trusted by default.
-  nix.settings.trusted-users = [ "root" "@wheel" ];
-
-  security.sudo.wheelNeedsPassword = false;
-
-  # Enable SSH everywhere
-  services.openssh.enable = true;
-
-  # No need for sound on a server
-  sound.enable = false;
-
-  # UTC everywhere!
-  time.timeZone = lib.mkDefault "UTC";
-
-  # No mutable users by default
-  users.mutableUsers = false;
-
-  systemd = {
-    # Given that our systems are headless, emergency mode is useless.
-    # We prefer the system to attempt to continue booting so
-    # that we can hopefully still access it remotely.
-    enableEmergencyMode = false;
-
-    # For more detail, see:
-    #   https://0pointer.de/blog/projects/watchdog.html
-    watchdog = {
-      # systemd will send a signal to the hardware watchdog at half
-      # the interval defined here, so every 10s.
-      # If the hardware watchdog does not get a signal for 20s,
-      # it will forcefully reboot the system.
-      runtimeTime = "20s";
-      # Forcefully reboot if the final stage of the reboot
-      # hangs without progress for more than 30s.
-      # For more info, see:
-      #   https://utcc.utoronto.ca/~cks/space/blog/linux/SystemdShutdownWatchdog
-      rebootTime = "30s";
+  config = {
+    # List packages installed in system profile.
+    environment.systemPackages = map lib.lowPrio [
```

Comment:
Since this change these packages aren't installed?

Reply:
I just checked and it looks like it's the case, but I'm unsure why.